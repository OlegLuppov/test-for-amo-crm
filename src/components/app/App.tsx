import React from 'react'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import './app.scss'
import { NavBar } from '../nav_panel/NavBar'
import { TaskOne } from '../pages/task_one/TaskOne'
import { TaskTwo } from '../pages/task_two/TaskTwo'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <div className="container">
        <NavBar />
        <Routes>
          <Route element={<TaskOne />} path="task_1" />
          <Route element={<TaskTwo />} path="task_2" />
        </Routes>
      </div>
    </BrowserRouter>
  )
}

export default App
