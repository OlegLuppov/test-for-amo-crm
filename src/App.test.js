import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import App from './components/app/App'

describe('TEST APP', () => {
	render(<App />)

	test('render buttons and input Secons', () => {
		const buttonTaskOne = screen.getByText(/task_1/i)
		const buttonTaskTwo = screen.getByText(/task_2/i)
		const buttonStart = screen.getByText(/Start/i)
		const inputSeconds = screen.getByPlaceholderText(/Seconds/i)

		expect(buttonTaskOne).toBeInTheDocument()
		expect(buttonTaskTwo).toBeInTheDocument()
		expect(buttonStart).toBeInTheDocument()
		expect(inputSeconds).toBeInTheDocument()

		expect(inputSeconds).toMatchSnapshot()
	})

	test('No elements', () => {
		const helloWorld = screen.queryByText(/Hello World/i)
		expect(helloWorld).toBeNull()
	})
})
