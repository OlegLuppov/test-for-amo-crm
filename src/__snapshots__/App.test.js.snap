// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP render buttons and input Secons 1`] = `
<input
  placeholder="Seconds"
  type="text"
  value=""
/>
`;

exports[`render buttons and input Secons 1`] = `
<input
  placeholder="Seconds"
  type="text"
  value=""
/>
`;
